packages/pyright-internal/src/common/uri/uri.ts

```diff
@@ -0,0 +1,206 @@
+/*
+ * uri.ts
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ *
+ * URI namespace for storing and manipulating URIs.
+ */
+
+import { URI, Utils } from 'vscode-uri';
+import { EmptyUri } from './emptyUri';
+import { FileUri } from './fileUri';
+//import * as memoization from './memoization_instrumenting';
+import { WebUri } from './webUri';
+
+export interface Uri {
+    // Unique key for storing in maps.
+    readonly key: string;
+
+    // Returns the scheme of the URI.
+    readonly scheme: string;
+
+    // Returns the last segment of the URI, similar to the UNIX basename command.
+    readonly basename: string;
+
+    // Returns the extension of the URI, similar to the UNIX extname command.
+    readonly extname: string;
```

Comment:
I noticed that `extname` returns ".ext". Should it return "ext"? That might be a religious battle. I don't feel strongly either way, but I think it should be documented here.

Reply:
I'll add a comment.